erlang/README.md

```diff
@@ -0,0 +1,170 @@
+case you have alternate needs then either update Makefile or bake
+your image on the shell. See the example below as an example, where
+the target is x86_64 (change it appropriate to your requirement).
+
+```
+rumpbake hw_virtio beam.bin build/bin/x86_64-rumprun-netbsd/beam
+```
+
+Setup the networking as follows (if not done already).
+
+```
+sudo ip tuntap add tap0 mode tap
+sudo ip addr add 10.0.120.100/24 dev tap0
+sudo ip link set dev tap0 up
+```
+
+You are now ready to run your first erlang unikernel. To run the rumpkernel using qemu, the following will work:
+
+```
+ERLAPP_PATH=/apps/erlang
+ERLHOME=/tmp
+ERLPATH=/opt/erlang
+rumprun qemu \
+   -I if,vioif,'-net tap,script=no,ifname=tap0' \
+   -W if,inet,static,10.0.120.101/24 \
+   -e ERL_INETRC=$ERLPATH/erl_inetrc \
+   -b images/erlang.iso,$ERLPATH \
+   -b examples/app.iso,$ERLAPP_PATH \
+   -i beam.hw.bin \
+     "-- -no_epmd -root $ERLPATH/lib/erlang -progname erl -- -home $ERLHOME -noshell -pa $ERLAPP_PATH -s echoserver start"
+```
+
+You should see `Echo server started` text printed to your screen.
+
+Now you can connect to the server via telnet as shown below:
+
+```
+$ telnet 10.0.120.101 8080
+Trying 10.0.120.101...
+Connected to 10.0.120.101.
+Escape character is '^]'.
+GET /
+GET /
+hello there!
+hello there!
+^]
+
+telnet> quit
+Connection closed.
+```
+
+You should replace `qemu` with `kvm` to run under kvm. For xen there will be
+equivalent variation and additionally use xen baked unikernel as well.
+
+Customizations
+==============
+
+The /opt/erlang
+---------------
+
+In case you want to change the /opt/erlang path then
+you'll have to change the Makefile, erl_inetrc and all
+the manual instructions mentioned in this document.
+
+Notes
+=====
+
+DNS
+---
+
+At present the erlang configuration inet configuration is tuned
+to use file and DNS as a source of resolving ip/host, though untested.
+
+Shortcomings
+============
+
+EPMD (clustering)
+-----------------
+
+In it's current form the epmd is not integrated within the baked
+image, so there is no support for clustering. Having said that it is
+not very difficult to hack together an image and get it working, but
+this is in the works at the time of this writing.
+
+
+Unoptimized Memory Allocation
+-----------------------------
+
+At present there is an unoptimal allocation of memory as a consequence of
+erlang alloc interacting with rumprun. This will be fixed in the near
+future, but something to be aware of in case you care. The exact amount
+of memory wasted is not quantified, because it needs to be fixed anyways.
+
+SMP (May not work)
+------------------
+
+> SMP image can be built but will not work at present.
+
+In case you want to build smp version the run the following command:
+
+```
+make beam.smp.hw.bin
+```
+
+Note that although you can build the unikernel for smp, but its not gonna work.
+This needs to be worked out before declaring for general use.
```

Comment:
There is no SMP in Rumprun, and I don't think there ever should be.
(nb. rump kernels support SMP, but Rumprun doesn't)